src/details/ArborX_DetailsFDBSCAN.hpp

```diff
@@ -22,14 +23,35 @@ namespace ArborX
 namespace Details
 {
 
+template <typename MemorySpace>
+struct FDBSCANNumNeighEarlyExitCallback
```

Comment:
I'd like the name to indicate a) that it is counting neighbors, b) that it may terminate early, c) that it is part of FDBSCAN. I don't know if there's a better name having those features.

Reply:
`CountN_<Andrey-specific suffix>`

Reply:
`FDBSCANCountNeighborsEarlyExitCallback`?

Reply:
There is nothing specific to FDBSCAN.  Early exit is irrelevant.

Reply:
> There is nothing specific to FDBSCAN. Early exit is irrelevant.

No, early exit is very relevant. The result may not count all the neighbors, and is the optimization one has to be aware of when reading FDBSCAN code. Thus, it is specific to DBSCAN (as it uses minpts).

Reply:
And you will notice the name say "count n" not "count"

Reply:
I agree with that @dalg24 that the structure itself is not specific to FDBSCAN even if right now that's the only place it can be used. I don't have a strong opinion for the rest of the name. I dislike both options equally :smile:

Reply:
I think it would be better to use a `(F)DBSCAN` namespace if we want to use it elsewhere later we can always move. Since it's currently internal that should not be a big concern.

Reply:
So, replace Details namespace by FDBSCAN and remove FDBSCAN from callbacks' names? Does that sound reasonable to everybody?

Reply:
I think I also would prefer dropping "early exit" since it's an implementation default.

Reply:
But we should clearly describe that there is an upper limit. I didn't catch easily `CountN` to describe "up to n" so would prefer something more descriptive like `CountWithUpperLimit` or so.
But then I don't have a strong preference after dropping "early exit" and the prefix from the name.

Reply:
OK, decided on the name `CountUpToN`.